development/extensions/tutorial_testing.rst

```diff
@@ -806,177 +805,488 @@ the following two help topics:
+your unit, database and functional tests will be executed.
 
-.. warning::
-
-    You should not edit the content of this file!
+Customising Your Test Workflow
+------------------------------
 
-.. code-block:: bash
+You'll notice the ``env`` section of the workflow has options where you can enable or
+disable basic tests such as the EPV, Code Sniffer, checking any images in your extension
+for ICC profiles and checking for executable text files by changing their variables to
+either 1 or 0.
 
-    #!/bin/bash
-    #
-    # This file is part of the phpBB Forum Software package.
-    #
-    # @copyright (c) phpBB Limited <https://www.phpbb.com>
-    # @license GNU General Public License, version 2 (GPL-2.0)
-    #
-    # For full copyright and license information, please see
-    # the docs/CREDITS.txt file.
-    #
-    set -e
-    set -x
+You can also set the version of phpBB you want your extension tested in using the
+``PHPBB_BRANCH`` variable. This tutorial is using the current version of phpBB which
+is the 3.3.x branch. Older branches of phpBB are not being supported with Github Actions,
+but can instead be used with `Travis-CI.com <https://travis-ci.com>`_ .
 
-    BRANCH=$1
+Finally you may also make changes to the jobs. This tutorial's workflow is split up into several
+jobs based on the databases being tested:
 
-    # Copy extension to a temp folder
-    mkdir ../../tmp
-    cp -R . ../../tmp
-    cd ../../
+* Basic Checks: The ``basic-checks`` job does not do any PHPUnit testing. This is where EPV, Code Sniffer and Image Profile checks run. If you never intend to run these checks you may either delete or comment out the entire ``basic-checks`` job.
 
-    # Clone phpBB
-    git clone --depth=1 "git://github.com/phpbb/phpbb.git" "phpBB3" --branch="$BRANCH"
-
-.. note::
+* MySQL Tests: The ``mysql-tests`` job runs PHPUnit tests in various MySQL and MariaDB and PHP combinations. They are all defined in the ``matrix`` section. This workflow only runs in versions of PHP 7 and PHP 8 but you could, for example, add tests for versions of PHP 5. (Note that to include PHP 5 in the test matrix you must change the ubuntu server for that job to ``runs-on: ubuntu-16.04``). You may delete or comment out some of the tests in the matrix if you do not want to test certain versions of PHP or MySQL or MariaDB.
 
-    The prepare-phpbb.sh file needs to have executable permissions or Travis CI
-    tests will fail. You can set the correct permission for this file from a
-    terminal command line interface, e.g.
+* PostgreSQL Tests: The ``postgres-tests`` job runs PHPUnit tests in various versions of PostgreSQL. The PHP version is consistent throughout since the mysql-checks job is the where we do most of our PHP environment checks. You may add additional checks to this matrix or you may either delete or comment out the entire ``postgres-tests`` job if you do not need to test PostgreSQL.
```

Comment:
```suggestion
* PostgreSQL Tests: The ``postgres-tests`` job runs PHPUnit tests in various versions of PostgreSQL. The PHP version is consistent throughout since the mysql-checks job is where we do most of our PHP environment checks. You may add additional checks to this matrix or you may either delete or comment out the entire ``postgres-tests`` job if you do not intend to test PostgreSQL.
```